jekyll/rails-addon.markdown

```diff
@@ -0,0 +1,152 @@
+---
+layout: default
+title: Addon for Rails
+nav_order: 10
+---
+
+# Ruby LSP Rails
+
+[Ruby LSP Rails](https://github.com/Shopify/ruby-lsp-rails) is a Ruby LSP addon that enhances the base Ruby LSP functionality
+with Rails-specific features. It provides improved code navigation, document symbols for Rails-specific elements,
+and runtime inspection capabilities.
+
+It [communicates with a running Rails instance](#runtime-inspection) to provide dynamic information about the application,
+enabling more accurate and context-aware language server features.
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Runtime Inspection](#runtime-inspection)
+- [Features](#features)
+    - [**Document Symbol**](#document-symbol)
+        - [Active Record Callbacks, Validations, and Associations](#active-record-callbacks-validations-and-associations)
+        - [Active Support Test Cases](#active-support-test-cases)
+    - [**Go to Controller Action Route**](#go-to-controller-action-route)
+    - [**Go to Controller Action View**](#go-to-controller-action-view)
+    - [**Go to Definition**](#go-to-definition)
+        - [Go to Active Record Callback and Validation Definitions](#go-to-active-record-callback-and-validation-definitions)
+        - [Go to Active Record Associations](#go-to-active-record-associations)
+        - [Go to Route Helper Definitions](#go-to-route-helper-definitions)
+    - [**Ruby File Operations**](#ruby-file-operations)
+        - [Commands](#commands)
+    - [**Run and Debug**](#run-and-debug)
+        - [Run Tests With Test Explorer](#run-tests-with-test-explorer)
+        - [Run Tests In The Terminal](#run-tests-in-the-terminal)
+        - [Debug Tests With VS Code](#debug-tests-with-vs-code)
+
+## Installation
+
+{: .important }
+> The Rails addon is installed automatically.
+
+Ruby LSP detects Rails projects and installs the [Rails addon](https://github.com/Shopify/ruby-lsp-rails) for you.
+
+## Runtime Inspection
+
+LSP tooling is typically based on static analysis, but `ruby-lsp-rails` actually communicates with your Rails app for
+some features.
+
+When Ruby LSP Rails starts, it spawns a `rails runner` instance which runs
+[`server.rb`](https://github.com/Shopify/ruby-lsp-rails/blob/main/lib/ruby_lsp/ruby_lsp_rails/server.rb).
+The addon communicates with this process over a pipe (i.e. `stdin` and `stdout`) to fetch runtime information about the application.
+
+When extension is stopped (e.g. by quitting the editor), the server instance is shut down.
+
+## Features
+
+### **Document Symbol**
```

Comment:
Because this theme's `####` is not easy to read, I use `### **txt** as header 3 and `### txt` as header 4.

We can either improve the styling or the way headers are organized later.